pkg/vsphere/toolbox/trace_channel.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package toolbox
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+)
+
+var (
+	Trace = true // TODO: make optional
```

Comment:
Instead of doing that maybe you could add two files one with functions doing nothing and have a build flag 
```
// +build !trace
...
func (d *TraceChannel) Send(buf []byte) error {
    return nil
}
...
```
and another with 
```
// +build trace
...
+func (d *TraceChannel) Send(buf []byte) error {
 +	if len(buf) > 0 {
 +		fmt.Fprintf(d.log, "SEND %d...\n%s\n", len(buf), hex.Dump(buf))
 +	}
 +
 +	err := d.Channel.Send(buf)
 +
 +	return err
 +}
```

and enable disable tracing with a build tag (go build -tags trace ...) and call those functions without decorating if statements?